internal/game/game.go

```diff
@@ -115,6 +116,44 @@ func (board *BoardState) Clone() *BoardState {
 	return clonedBoard
 }
 
+// Check that the provided move is legal.
+// A move is illegal when the pawn only moves to an adjacent cell and not further.
+func (board *BoardState) CheckMoveLegality(from CellIdentifier, to CellIdentifier) error {
+	// Compute the column diff of the move.
+	columnDiff := math.Abs(float64(from.Column - to.Column))
+	// Compute the row diff of the move.
+	rowDiff := math.Abs(float64(from.Row - to.Row))
+
+	if columnDiff+rowDiff != 1 {
+		// The move is illegal (more than 1 difference, or no difference).
+		if rowDiff == 1 && columnDiff == 1 {
+			// Detected a diagonal move, return a specific error.
+			return errors.New("a pawn cannot move in diagonal")
+		} else {
+			// General error.
+			return fmt.Errorf("'%s' cannot be reached from '%s'", to.String(), from.String())
+		}
+	} else {
+		// The move is legal.
+		return nil
+	}
```

Comment:
This is definitly easier to understand!

**thought:** Maybe early returns would make the code even more easier to understan

```suggestion
	if columnDiff+rowDiff == 1 {
		return nil
	}
	
	// The move is illegal (more than 1 difference, or no difference).
	if rowDiff == 1 && columnDiff == 1 {
		// Detected a diagonal move, return a specific error.
		return errors.New("a pawn cannot move in diagonal")
	}
	
	return fmt.Errorf("'%s' cannot be reached from '%s'", to.String(), from.String())
```